topics/reference_guide/custom_language_support/language_server_protocol.md

```diff
@@ -0,0 +1,175 @@
+<!-- Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+# Language Server Protocol
+
+<link-summary>Language Server Protocol support in IntelliJ-based IDEs</link-summary>
+
+The [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) is an open-standard protocol developed by Microsoft. It enables communication between development tools and Language Servers.
+Language Servers can provide language-specific features such as code completion, documentation, and formatting, which is far easier than implementing language support from scratch.
+It also reduces the need for constant maintenance and tracking of changes in relevant languages and tools, making it easier to bring consistent language support to various development environments.
+
+However, the canonical custom language support provided by IntelliJ Platform still offers a wider range of integration with IDE features than handling and presenting data provided by a Language Server.
```

Comment:
link https://plugins.jetbrains.com/docs/intellij/custom-language-support.html for "canonic..."